plugins/parsers/value/parser.go

```diff
@@ -23,6 +23,7 @@ func (v *ValueParser) Parse(buf []byte) ([]telegraf.Metric, error) {
 	if v.DataType == "string" {
 		vStr = strings.TrimSpace(string(buf))
 	} else {
+		buf = bytes.Trim(buf, "\x00")
```

Comment:
Though I think maybe we should trim whitespace from none string values as well, In which case we could call bytes.TrimSpace and bytes.Trim(buf, "\x00") first thing in the function